### DISCUSSION TITLE
function returns unexpected value when is called via .rpc

### DISCUSSION THREAD

**Discussion by juanbits** [supabase/supabase#15788] [Questions] [answered]
hi everyone, 

i create a custom function check_info(my_arg), this could return 2 possible values -1 or 1

when i send the `11432db7-4a85-4ce7-b72c-0b0dee965d23` as `my_arg` value, the expected result is 1

if i call this function in the SQLEditor tab, i got the 1 value
![image](https://github.com/supabase/supabase/assets/5168033/c65a49cf-cc72-4adf-a991-74d561fef0f5)

but if i call this via 
`supabase.rpc('check_info', {'my_arg', 'value'}`
![image](https://github.com/supabase/supabase/assets/5168033/2fb45e56-fab5-42ff-a1e9-b6a5e6da7f57)

i got the -1 value
![image](https://github.com/supabase/supabase/assets/5168033/52cf1eae-b454-4795-9597-fb2f0e33cd44)

why happens this behavior?

thanks in advance for your help!

**Comment by juanbits** [comment, marked as answer]
resolved. the fix was enable RLS in the involved tables

**Reply by javogc** [reply]
I created a SELECT RLS policy for my table (let's call it table A), but I'm still getting a null in the client. I have another .rpc function, querying another table B, and that one works. What's weird is that if I swap the table's value on the .rpc function that works to table A, then it returns the expected value. ¿Any thoughts on what might be happening?
